osu.Game/Storyboards/StoryboardSprite.cs

```diff
@@ -209,24 +203,104 @@ private interface IGeneratedCommand
+                                                           .Loop(command.Delay, command.LoopCount);
+                            }
+                        }
+
+                        break;
+
+                    case "FlipH":
+                    case "FlipV":
+                        if (command.LoopCount == 0)
+                        {
+                            using (drawable.BeginAbsoluteSequence(command.StartTime))
+                            {
+                                ((IFlippable)drawable).TransformTo(command.PropertyName, command.StartValue).Delay(command.Duration)
+                                                      .TransformTo(command.PropertyName, command.EndValue);
+                            }
+                        }
+                        else
+                        {
+                            using (drawable.BeginAbsoluteSequence(command.StartTime))
+                            {
+                                ((IFlippable)drawable).TransformTo(command.PropertyName, command.StartValue).Delay(command.Duration)
+                                                      .TransformTo(command.PropertyName, command.EndValue)
+                                                      .Loop(command.Delay, command.LoopCount);
+                            }
+                        }
+
+                        break;
+
+                    case "Blending":
+                        if (command.LoopCount == 0)
+                        {
+                            using (drawable.BeginAbsoluteSequence(command.StartTime))
+                            {
+                                drawable.TransformTo(command.PropertyName, command.StartValue).Delay(command.Duration)
+                                        .TransformTo(command.PropertyName, command.EndValue);
+                            }
+                        }
+                        else
+                        {
+                            using (drawable.BeginAbsoluteSequence(command.StartTime))
+                            {
+                                drawable.TransformTo(command.PropertyName, command.StartValue).Delay(command.Duration)
+                                        .TransformTo(command.PropertyName, command.EndValue)
+                                        .Loop(command.Delay, command.LoopCount);
+                            }
+                        }
+
+                        break;
+
+                    default:
+                        if (command.LoopCount == 0)
+                        {
+                            using (drawable.BeginAbsoluteSequence(command.StartTime))
+                            {
+                                drawable.TransformTo(command.PropertyName, command.StartValue).Then()
+                                        .TransformTo(command.PropertyName, command.EndValue, command.Duration, command.Easing);
```

Comment:
This is not entirely correct because of the concept of "grouping". Take position transforms for example, `MoveToX`/`MoveToY` supply a "grouping" property (`Drawable.Position`) when they call `TransformTo`.

I'm in favour of the previous code, but change the delegate to return a `TransformSequence` so you can apply the loop command on top of it.

Reply:
Hmm, about `MoveToX/Y`: will there be a difference in the end?
Imo this code is way more simpler to understand (though with some changes needed) and keep track of all the delays and offsets applied, but sure I'll check what I can do.